### PR TITLE
[diffusion] Refactor ComfyUI integrated mode onto native pipelines + adapters

### PR DESCRIPTION
**Pull request by niehen6174**
## Summary

ComfyUI integrated mode is no longer a third copy of each diffusion pipeline. It is a **DiT-only runtime profile** on the native Flux / Z-Image / Qwen pipelines (`--comfyui-mode`).

- Drop the three `comfyui_*_pipeline.py` forks (~1450 lines).
- Keep one shared executor. Model differences live in a **checkpoint spec** (single-file `.safetensors` remap) and a **step adapter** (`pack` / `unpack` / `fill_req`).
- Keep the existing process topology (ComfyUI process + spawned workers). Local hops use **CUDA IPC handles** instead of pickling GPU tensors through host memory. Multi-rank `--comfyui-mode` detaches CUDA tensors and broadcasts them over **NCCL**. Conditioning is cached in a **run-scoped worker session**.

Adding another image DiT that already has ComfyUI UNET / CLIP / VAE / KSampler should be two small files, not another 400–700 line pipeline.

## Motivation

Integrated mode exists so ComfyUI keeps CLIP, VAE, the sampler loop, CFG, and community nodes, while SGLang only accelerates the DiT. That boundary is unchanged.

The old implementation encoded that boundary by **copying a pipeline and an executor per model**. Flux / Z-Image / Qwen already had native pipelines. The ComfyUI copies existed only because:

1. ComfyUI gives a single `.safetensors` UNET, not `model_index.json`.
2. The worker must run one DiT step with a pass-through scheduler.
3. ComfyUI key names do not match SGLang `param_names_mapping`.

`comfyui_mode` was already scattered through `DenoisingStage` / `SamplingParams`, but it was never a first-class profile, so each new model copied the skeleton again.

| Layer | Before | Actually model-specific |
| --- | --- | --- |
| `ComfyUIXxxPipeline` | ~1450 lines across 3 files | weight mapping, QKV split, scale/shift order |
| `XxxExecutor` | ~300 lines across 4 classes | pack / unpack, timestep scale, which `Req` fields get embeds |
| `generator.py` tables | edit two dicts per model | none |

A new image DiT needed ~500 lines of glue. Most of it was identical.

## Architecture
- **Befor** (loop, CFG, `model_options` callables, community nodes) stays in the ComfyUI process.
- **After** (DiT, SP/TP, TeaCache / BCG / FSDP) stays in the SGLang workers.
- **Between** them: tensors only (IPC-able). No Python closures across the process boundary.

comfyui_sglang_refactor

No new `comfyui_*_pipeline.py`. `pipeline_class_dict` / `executor_class_dict` are generated from `adapter.model_types`.

## Why this is easier to extend

| | Before | After |
| --- | --- | --- |
| New image DiT | pipeline 400–700 + executor 70–170 + two registries | **checkpoint spec + adapter** (Qwen spec ~55, Z-Image ~63, typical adapter 40–90) |
| Weight mapping | ComfyUI copy + native config, easy to diverge | one `ComfyUICheckpointSpec`; `inherit_config_mapping` when the two mappings share a source namespace |
| Loader features | Flux used a 175-line handwritten `state_dict` loop (no FSDP / quant) | shared `TransformerLoader` / `weights_iterator` (same shape as GGUF) |
| H3 / packed video | executor only understood `forward(x, t, context)` + 4D image pack | `PackedForward.extra_req` and nested latents are reserved; H3 adapter is **not** in this PR |
| Server mode | untouched | untouched |

Concrete correctness win from making the profile real: the old Flux ComfyUI loader **merged** native `FluxConfig.param_names_mapping` (targets like `to_qkv`, `ff.linear_in`) with the ComfyUI aliases. Mapping is applied to a fixpoint, so already-correct diffusers names were rewritten to names the live module does not have. The handwritten loader built on a real device and left unmatched params **randomly initialized** with a warning. The new path builds on meta device and errors instead, which is how the bug surfaced. Flux now sets `inherit_config_mapping=False`. Z-Image keeps `True` because its native mapping must merge split `to_q/k/v` back into fused `to_qkv`.

The adapter interface is sized for H3 (nested latents, structured payload), not for Flux's three-tensor case. This PR does **not** add MiniMax-H3 support; it stops the next model from needing another pipeline fork.

## Transport / performance

These hops were using the **per-request** distribution path on every **sampler step**.

| Hop | Before | After |
| --- | --- | --- |
| ComfyUI ↔ rank 0 | `pickle.dumps` copies CUDA tensors to host. Measured: **460 KB tensor → 461 KB pickle** | `CudaIpcRef` handle. Unit test: handle pickle **< 2 KB**. Receiver `_new_shared_cuda` + `clone` |
| Multi-rank `--comfyui-mode` | whole `Req` over `broadcast_pyobj` / gloo CPU, then `_fix_tensor_device` | control envelope still gloo; GPU tensors via existing NCCL `broadcast_tensor_dict`. General SP / CFG / TP `recv` is **unchanged** |
| Conditioning | CLIP / pooled / image embeds resent every step | `comfyui_session_id`; later steps send latent + timestep |

On current image models this is **not** a large wall-clock win. Flux 1024² packed latent is ~0.5 MB/step, so host pickle was already cheap next to DiT (~124 ms Flux / ~232 ms Qwen). The leftover ~10 ms/step is mostly scheduler wakeup, not the tensor copy.

The transport work is for correctness and the next model: stay off host pickle / gloo, keep GPU tensors as IPC / NCCL handles, and cache conditioning so later steps do not resend embeds. A video DiT (H3 ~14 MB/step) would otherwise copy gigabytes over a full sampler. Processes stay separate on purpose — folding ComfyUI into rank-0 would shave that last ~10 ms, but mixes its event loop with NCCL and process-global CUDA state.

## Test plan

- [x] `test/unit/test_comfyui_profile.py` — profile assembly + Flux synthetic single-file load (meta device, fused QKV / adaLN)
- [x] `test/unit/test_comfyui_adapters.py` — pack / unpack
- [x] `test/unit/test_comfyui_session.py` — conditioning cache
- [x] `test/unit/test_ipc_cuda.py` — handle pickle < 2 KB, cross-process roundtrip
- [x] Weight convert equivalence vs the deleted pipelines on synthetic state dicts (Flux / Z-Image / Qwen names, merge info, values)
- [x] Z-Image `comfyui_mode` one-step (`noise_pred` `[1, 16, 1, 90, 160]`) and ComfyUI-format single-file load (6.15B)
- [x] Flux `comfyui_mode` 2-GPU one-step (`noise_pred` `[1, 3600, 64]`, ~0.7 s)
- [ ] Full ComfyUI UI / KSampler workflow on a machine with matching `comfy_aimdo` / `comfy_kitchen`
- [ ] Existing plugin e2e (`test_flux_pipeline.py`, `test_zimage_pipeline.py`, Qwen) on real checkpoints

---
### CI States

Latest PR Test (Base): :x: [Run #33407655401](https://github.com/sgl-project/sglang/actions/runs/33407655401)
Latest PR Test (Extra): :x: [Run #33407655389](https://github.com/sgl-project/sglang/actions/runs/33407655389)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33407655523](https://github.com/sgl-project/sglang/actions/runs/33407655523)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/distributed/ipc_cuda.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/distributed/ipc_cuda.py:1]: maybe move it to `distributed/utils`?

Comment (niehen6174) [reply python/sglang/multimodal_gen/runtime/distributed/ipc_cuda.py:1]: Moved it under distributed/  instead of folding it into utils.py.
That file gets star-imported by the package, so dumping the CUDA IPC retain/spill path in there would load it on every distributed import.

**Review by mickqian — COMMENTED** [review]
Current-head CI root failure: test_declared_alias_loads_by_exact_key_and_structural_source passes a SimpleNamespace without comfyui_mode to is_comfyui_mode, which directly reads that field: https://github.com/sgl-project/sglang/actions/runs/33407655401/job/99539408863 . Align the fixture with the supported argument contract and validate that real loader test. Component/B200/multi-GPU failures are health-check cascades. The 14:30 UTC snapshot also reported main conflicts (latest query UNKNOWN). I do not have fork push access; please resolve and push before final CI.
